azurelinuxagent/ga/monitor.py

```diff
@@ -399,54 +409,38 @@ def send_telemetry_heartbeat(self):
 
             self.last_telemetry_heartbeat = datetime.datetime.utcnow()
 
-    def send_cgroup_telemetry(self):
-        if self.last_cgroup_telemetry is None:
-            self.last_cgroup_telemetry = datetime.datetime.utcnow()
-
-        if datetime.datetime.utcnow() >= (self.last_telemetry_heartbeat + MonitorHandler.CGROUP_TELEMETRY_PERIOD):
-            try:
-                metric_reported, metric_threshold = CGroupsTelemetry.collect_all_tracked()
-                for cgroup_name, metrics in metric_reported.items():
-                    thresholds = metric_threshold[cgroup_name]
-
-                    for metric_group, metric_name, value in metrics:
-                        if value > 0:
-                            report_metric(metric_group, metric_name, cgroup_name, value)
-
-                        if metric_group == "Memory":
-                            # Memory is collected in bytes, and limit is set in megabytes.
-                            if value >= CGroupConfigurator._format_memory_value('megabytes', thresholds.memory_limit):
-                                msg = "CGroup {0}: Crossed the Memory Threshold. " \
-                                      "Current Value: {1} bytes, Threshold: {2} megabytes." \
-                                       .format(cgroup_name, value, thresholds.memory_limit)
-
-                                logger.warn(msg)
-                                add_event(name=AGENT_NAME,
-                                          version=CURRENT_VERSION,
-                                          op=WALAEventOperation.CGroupsLimitsCrossed,
-                                          is_success=True,
-                                          message=msg,
-                                          log_event=True)
-
-                        if metric_group == "Process":
-                            if value >= thresholds.cpu_limit:
-                                msg = "CGroup {0}: Crossed the Processor Threshold. " \
-                                      "Current Value: {1}, Threshold: {2}." \
-                                       .format(cgroup_name, value, thresholds.cpu_limit)
-
-                                logger.warn(msg)
-                                add_event(name=AGENT_NAME,
-                                          version=CURRENT_VERSION,
-                                          op=WALAEventOperation.CGroupsLimitsCrossed,
-                                          is_success=True,
-                                          message=msg,
-                                          log_event=True)
-
-            except Exception as e:
-                logger.warn("Monitor: failed to collect cgroups performance metrics: {0}", ustr(e))
-                logger.verbose(traceback.format_exc())
-
-            self.last_cgroup_telemetry = datetime.datetime.utcnow()
+    def send_extension_metrics_telemetry(self):
```

Comment:
minor: can we separate this into two functions (poll and report) and call both from the main loop (makes it a little clearer that they are separate operations with different periods)

Reply:
When you say - 

> call both from the main loop

With the main loop, do you mean to add another method call only for polling, in monitor.py#270? I wanted the method to be self-contained with the task of handling all the extension_metrics related operations.
```
while self.should_run:
    ...
    self.send_extension_metrics_telemetry()
    ...
```

I have tried to simplify the `send_extension_metrics_telemetry` method further. Let me know if that works. We can discuss this as well.

Reply:
Since it is 2 different operations (collect and send) with different periods I think having 2 calls from the main loop makes the intention clearer